File: BuildingPrototype/BuildingPrototype/Data/GlobalSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;

namespace MapEditor.Data
{
    static class GlobalSettings
    {
        private static Vector2 resolution;
        private static int tileSize;
        private static float scale = 1.0f;

        public static Vector2 Resolution
        {
            get { return GlobalSettings.resolution; }
            set { GlobalSettings.resolution = value; }
        }
        public static float Scale
        {
            get { return GlobalSettings.scale; }
            set { GlobalSettings.scale = Math.Max(0.1f, value); }
        }

        public static int ScaledTileSize
        {
            get { return Math.Max(1, (int)(scale * GlobalSettings.tileSize)); }
        }
        public static int TileSize
        {
            get { return GlobalSettings.tileSize; }
            set { GlobalSettings.tileSize = value; }
        }
        public static Vector2 TileResolution
        {
            get { return resolution / ScaledTileSize; }
        }
    }
}

File: JavaPort/src/data/GlobalSettings.java
package data;

import org.lwjgl.util.vector.Vector2f;

public class GlobalSettings {
	private static Vector2 resolution;
	private static int tileSize;
	private static float scale = 1.0f;

	public static Vector2 getResolution() {
		return resolution;
	}

	public static void setResolution(Vector2 resolution) {
		GlobalSettings.resolution = resolution;
	}

	public static int getTileSize() {
		return tileSize;
	}

	public static void setTileSize(int tileSize) {
		GlobalSettings.tileSize = tileSize;
	}

	public static float getScale() {
		return scale;
	}

	public static void setScale(float scale) {
		GlobalSettings.scale = Math.max(0.1f, scale);
	}

	public static int getScaledTileSize() {
		return Math.max(1, (int) (scale * GlobalSettings.tileSize));
	}

	public static Vector2 getTileResolution() {
		return Vector2.vectorScale(resolution, 1.0f / getScaledTileSize());

	}
}